src/Rootfs.jl

```diff
@@ -57,6 +59,11 @@ function Base.:(==)(a::CompilerShard, b::CompilerShard)
            a.archive_type == b.archive_type
 end
 
+function Base.isless(a::CompilerShard, b::CompilerShard)
+    return isless((a.name, a.version, string(a.target), a.host, a.archive_type),
+                  (b.name, b.version, string(b.target), b.host, b.archive_type))
+end
```

Comment:
I don't like defining `isless` for `CompilerShard`s, this looks very arbitrary, but I presume you can define a custom function to be passed to the sorting functions when needed?

Reply:
Moved to a local `isless_shards` function where needed in 1ef1746.